pennylane/transforms/core/transform_program.py

```diff
@@ -167,6 +167,61 @@ def insert_front(self, transform_container: TransformContainer):
             )
         self._transform_program.insert(0, transform_container)
 
+    def add_transform(self, transform: TransformDispatcher, *targs, **tkwargs):
+        """Add a transform (dispatcher) to the end of the program.
```

Comment:
I think I'd wanna add something a bit more cautionary just in case, not sure if this suffices. I suspect if you're reading these docs you already know the difference, and if you don't... proceed with caution. You'll get an error anyway below (which is awesome). On the other hand, maybe the error is enough?
```suggestion
        """Add a transform (dispatcher) to the end of the program.

        Note that this should be a function decorated with/called by
        `qml.transforms.transform`, and not a `TransformContainer`.
```

Reply:
I don't mind adding a note :+1: